src/main/java/duke/commands/NewNoteCommand.java

```diff
@@ -0,0 +1,237 @@
+package duke.commands;
+
+import duke.notes.Note;
+import duke.notes.event.Event;
+import duke.notes.event.Birthday;
+import duke.notes.event.Wedding;
+import duke.notes.task.Bill;
+import duke.notes.task.Deadline;
+import duke.notes.task.Shoplist;
+import duke.notes.task.Task;
+import duke.parser.DateException;
+import duke.parser.DateParser;
+import duke.storage.DukeList;
+import duke.storage.DukeStorage;
+import duke.ui.DukeUI;
+import java.util.ArrayList;
+import java.util.Date;
+import java.text.ParseException;
+
+/**
+ * An extension of the {@code DukeCommand} object that evaluates user input to create new {@code Note} objects.
+ *
+ * @author tanqiuyu
+ * @since 2020-09-16
+ */
+public class NewNoteCommand extends DukeCommand {
+
+    //VARIABLES-----------------------------------------
+    private NoteType noteType;
+    private ArrayList<String> inputs;
+
+    //CONSTRUCTORS--------------------------------------
+    /**
+     * This method constructs a {@code NewNoteCommand} object.
+     *
+     * @param noteType The type of {@code Note} being created.
+     * @param inputs The list of user input elements to be used to create new {@code Note} objects.
+     */
+    public NewNoteCommand(String noteType, ArrayList<String> inputs) throws CommandException {
+        super("NEWNOTE");
+        this.noteType = NoteType.getKey(noteType);
+        this.inputs = inputs;
+    }
+
+    /**
+     * This method initialises a {@code NewNoteCommand} object.
+     */
+    @SuppressWarnings("unused")
+    public NewNoteCommand() { super(); }
+
+    //METHODS-------------------------------------------
+    /**
+     * This method checks for clashes between new and existing {@code Event} objects, as well as the current date-time.
+     *
+     * @param notes The {@code ArrayList} object that holds the notes managed by {@code Duke}.
+     * @param start The {@code Date} object indicating the start date of the new {@code Event} object.
+     * @param end The {@code Date} object indicating the end date of the new {@code Event} object.
+     * @exception DateException If there are errors in the formats or substance of {@code Date} objects.
+     */
+    static void checkForClashes(ArrayList<Note> notes, Date start, Date end)
+            throws DateException {
+
+        Date now = new Date();
+        if(start.before(now)){
+            throw new DateException(start, "StartB4Now");
+        }
+        if(end.before(start)){
+            throw new DateException(end, "EndB4Start");
+        }
+
+        for(Note note: notes){
+            if(note instanceof Event){
+                Date noteStart = ((Event) note).getStartDate();
+                Date noteEnd = ((Event) note).getEndDate();
```

Comment:
Can use guard clause to reduce nesting:
```java
for(Note note: notes){
    if(!note instanceof Event){
        continue;
    }
    Date noteStart = ((Event) note).getStartDate();
    Date noteEnd = ((Event) note).getEndDate();
    // ...
```

Reply:
Noted Prof, I've also looked thru the rest of the code for similar instances.